test/test_document.py

```diff
@@ -710,6 +710,19 @@ def test_write_validation(self):
         sbol2.Config.setOption(sbol2.ConfigOptions.VALIDATE, validate)
         sbol2.Config.setOption(sbol2.ConfigOptions.VERBOSE, verbose)
 
+    def test_append_string_no_overwrite(self):
+        doc = sbol2.Document(CRISPR_LOCATION)
+        result = doc.appendString(doc.writeString(), overwrite=False)
+        self.assertFalse(result)
+
+    def test_append_string(self):
+        doc = sbol2.Document()
+        cd = doc.componentDefinitions.create('cd1')
+        cd.components.create('c1')
+        self.assertEqual(1, len(cd.components))
+        doc.appendString(doc.writeString())
```

Comment:
To make the test more explicit and the inner workings of `append` more transparent, I suggest creating a second `Document` with a slightly modified object.
```
        doc = sbol2.Document()
        cd = doc.componentDefinitions.create('cd1')
        cd.components.create('c1')
        cd.roles = ['foo']
        self.assertEqual(1, len(cd.components))
        doc2 = sbol2.Document()
        cd_updated = doc2.componentDefinitions.create('cd1')
        cd_updated.roles = ['bar']
        doc.appendString(doc2.writeString())
        self.assertEqual(1, len(cd.components))
        self.assertEqual(cd.roles, ['bar'])
```

Reply:
After the `appendString` in the third to last line, shouldn't "cd1" have zero components? `doc.cd1` has been overwritten by `doc2.cd1`, and the latter has no child components. The test passes if I change the second to last line to:

```python
self.assertEqual(0, len(cd.components))
```

Reply:
I think my original intention was to clone the child component, but I left it out.  Doesn't matter, I like this test better.

Reply:
Or, it might be better for `cd_updated` to have a new component.  Then after roundtrip, check to see that the old child component was removed and replaced with the new one.  That seems like it would be a robust and comprehensive test.

Reply:
Adding a component to `cd_updated` will be part of the new test, with the appropriate checks added.

Reply:
Beautiful.

Reply:
This is now committed, as of https://github.com/SynBioDex/pySBOL2/pull/373/commits/0a9c30a98349202f2563134448c168cb060e3b24